### PR TITLE
1.19 Anti-Xray cleanup

### PR DESCRIPTION
**Pull request by stonar96**

### CODE REVIEW

File: patches/server/0004-Paper-config-files.patch [hunk 1]
```diff
@@ -1105,6 +1106,7 @@ index 0000000000000000000000000000000000000000..fa191f418079a8ee24326c5952d12d44
 +                    .register(DoubleOrDefault.SERIALIZER)
 +                    .register(BooleanOrDefault.SERIALIZER)
 +                    .register(Duration.SERIALIZER)
++                    .register(ChunkPacketBlockControllerAntiXray.EngineMode.SERIALIZER)
```
Comment (Machine-Maker) [inline patches/server/0004-Paper-config-files.patch]: The reason I added this in the X-ray patch, is that now the server won't compile if you only some patches before X-ray.

File: patches/server/0353-Anti-Xray.patch [hunk 2]
```diff
@@ -845,9 +845,7 @@ index 0000000000000000000000000000000000000000..a5d43901595f864c8a5dd5d013aa42d7
 +
 +        HIDE(1, "hide ores"),
 +        OBFUSCATE(2, "obfuscate");
-+
```
Comment (jpenilla) [inline patches/server/0353-Anti-Xray.patch:848]: personally I think the whitespace here improves readability

File: patches/server/0353-Anti-Xray.patch [hunk 3]
```diff
@@ -845,9 +845,7 @@ index 0000000000000000000000000000000000000000..a5d43901595f864c8a5dd5d013aa42d7
...
 +        public static final ScalarSerializer<EngineMode> SERIALIZER = new Serializer();
-+
```
Comment (jpenilla) [inline patches/server/0353-Anti-Xray.patch:850]: and here

File: patches/server/0353-Anti-Xray.patch [hunk 4]
```diff
@@ -1051,11 +1031,12 @@ index 568ac283baf40e2a89f5b002ffd899eba8008ef2..23d76eb74a88610472aa0288559efbaa
          this.heightmaps = new CompoundTag();
  
          for(Map.Entry<Heightmap.Types, Heightmap> entry : chunk.getHeightmaps()) {
-@@ -43,7 +46,13 @@ public class ClientboundLevelChunkPacketData {
+@@ -43,7 +46,14 @@ public class ClientboundLevelChunkPacketData {
          }
  
          this.buffer = new byte[calculateChunkSize(chunk)];
 -        extractChunkData(new FriendlyByteBuf(this.getWriteBuffer()), chunk);
++
```
Comment (jpenilla) [inline patches/server/0353-Anti-Xray.patch:983]: is this blank line intentional?

**Review by jpenilla — DISMISSED** [review]
Anywhere that has a note about "make sure this isn't used on update" should be annotated with `@DoNotUse` in addition to `@Deprecated` so that the project will fail to pass `check` when any Paper or Minecraft code calls it.

**Comment by bluelhf** [comment]
Does this PR account for new blocks with transparency, e.g. mangrove roots?

**Comment by stonar96** [comment]
> Does this PR account for new blocks with transparency, e.g. mangrove roots?

Any issues?

Edit: Indeed, fixed.

File: patches/server/0004-Paper-config-files.patch (hunk 1 above)
Comment (stonar96) [reply patches/server/0004-Paper-config-files.patch]: As discussed on Discord in #paper-contrib I have moved the `EngineMode` stuff to the config patch.

File: patches/server/0353-Anti-Xray.patch (hunk 2 above)
Comment (stonar96) [reply patches/server/0353-Anti-Xray.patch:848]: Moved to config patch and done.

File: patches/server/0353-Anti-Xray.patch (hunk 4 above)
Comment (stonar96) [reply patches/server/0353-Anti-Xray.patch:983]: Yes, I always use blank lines to separate code blocks that have indentation. This style is used in the entire patch.

**Review by electronicboy — APPROVED** [review]

**Review by kennytv — APPROVED** [review]

**Pull request merged**